libraries/base/GHC/Ptr.hs

```diff
@@ -165,7 +165,7 @@ instance Show (Ptr a) where
      where
         -- want 0s prefixed to pad it out to a fixed length.
        pad_out ls =
-          '0':'x':(replicate (2*32 - length ls) '0') ++ ls ++ rs
+          '0':'x':(replicate (2*8 - length ls) '0') ++ ls ++ rs
```

Comment:
Ahh because its a hex representation of the bytes of the address `Ptr` so 8 bytes on 64bit, with 2 0's for each byte (this is at least how haskell on 64bit machines do it) on 32bit this wont make sense as it should be smaller (4*2) (looks like GHC itself uses `SIZEOF_HSPTR`, but i don't think we have access to that?)